src/core/qgspointlocator.cpp

```diff
@@ -883,6 +882,36 @@ QgsPointLocator::Match QgsPointLocator::nearestEdge( const QgsPointXY &point, do
   return m;
 }
 
+QgsPointLocator::Match QgsPointLocator::nearestArea( const QgsPointXY &point, double tolerance, MatchFilter *filter )
+{
```

Comment:
Maybe worth checking whether the layer type is polygon and return invalid match immediately if it isn't?

Reply:
pointInPoint does it internally, so I don't think we're losing anything here